src/aws/auth.rs

```diff
@@ -1,16 +1,50 @@
 use std::time::Duration;
 
 use aws_config::{
-    default_provider::credentials::DefaultCredentialsChain, sts::AssumeRoleProviderBuilder,
+    default_provider::credentials::DefaultCredentialsChain, imds, sts::AssumeRoleProviderBuilder,
 };
 use aws_types::{credentials::SharedCredentialsProvider, region::Region, Credentials};
+use serde_with::serde_as;
 use vector_common::sensitive_string::SensitiveString;
 use vector_config::configurable_component;
 
 // matches default load timeout from the SDK as of 0.10.1, but lets us confidently document the
 // default rather than relying on the SDK default to not change
 const DEFAULT_LOAD_TIMEOUT: Duration = Duration::from_secs(5);
 
+/// IMDS Client Configuration for authenticating with AWS.
+#[serde_as]
+#[configurable_component]
+#[derive(Copy, Clone, Debug, Derivative)]
+#[derivative(Default)]
+#[serde(deny_unknown_fields)]
+pub struct ImdsAuthentication {
+    /// Number of IMDS retries for fetching tokens and metadata.
+    #[serde(default = "default_max_attempts")]
+    #[derivative(Default(value = "default_max_attempts()"))]
+    max_attempts: u32,
+
+    /// Connect timeout for IMDS.
+    #[serde(default = "default_timeout")]
+    #[serde_as(as = "serde_with::DurationSeconds<u64>")]
+    #[derivative(Default(value = "default_timeout()"))]
+    connect_timeout: Duration,
```

Comment:
Typically for duration config fields we've suffixed with the unit to make it easy to know what it expects without checking the docs. In this case, `connect_timeout_secs`. Same with `read_timeout`.

Reply:
🤔 that's fair - it's less noticeable when you looking at it here since it has the `DurationSeconds` in the annotations, but I can see that being a QOL improvement if you using it elsewhere in the code.

Reply:
Right, I'm less concerned about the code and more about users putting this in their config files 😄 

This is mentioned in https://github.com/vectordotdev/vector/blob/master/docs/specs/configuration.md#option-naming. Looking at that now, I'm noticing that it recommends using the full name for the unit, but I don't think we actually do that anywhere (preferring `secs` and `ms`). I think abbreviations are fine though, so I'll open a PR to update that.

Reply:
Scratch that, we do seem to be using `_seconds` too so let's do that here. I'll leave the doc as-is and we can start using `milliseconds` rather than `ms`.